spock-core/src/main/java/org/spockframework/runtime/SpockRuntime.java

```diff
@@ -34,21 +34,44 @@ public static void verifyCondition(@Nullable ValueRecorder recorder,
       @Nullable String text, int line, int column, @Nullable Object message, @Nullable Object condition) {
     if (!GroovyRuntimeUtil.isTruthy(condition)) {
       throw new ConditionNotSatisfiedError(
-          new Condition(getValues(recorder), text, TextPosition.create(line, column), messageToString(message)));
+          new Condition(getValues(recorder), text, TextPosition.create(line, column), messageToString(message), null, null));
     }
   }
 
+  public static final String CONDITION_FAILED_WITH_EXCEPTION = "conditionFailedWithException";
+
+  public static void conditionFailedWithException(@Nullable ValueRecorder recorder, @Nullable String text, int line, int column, @Nullable Object message, Throwable throwable){
+    if (throwable instanceof SpockAssertionError){
+      throw (SpockAssertionError) throwable; // this is our exception - it already has good message
+    }
+    if (throwable instanceof SpockException){
+      throw (SpockException) throwable; // this is our exception - it already has good message
+    }
+    throw new ConditionNotSatisfiedError(
```

Comment:
Yes and No.
It dependends why OOM occured. If it based on method parameters it will be good to provide parameters values to user thus he will be able to determine what is going wrong. Same for example for StackOverflowError. If OOM occures independently from test code than there is nothing we can do and does not matter when and where OOM occure again.